operator/internal/manifests/openshift/credentialsrequest.go

```diff
@@ -0,0 +1,96 @@
+package openshift
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/ViaQ/logerr/v2/kverrors"
+	cloudcredentialv1 "github.com/openshift/cloud-credential-operator/pkg/apis/cloudcredential/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/grafana/loki/operator/internal/manifests/storage"
+)
+
+const (
+	ccoNamespace = "openshift-cloud-credential-operator"
+)
+
+func BuildCredentialsRequest(opts Options) (*cloudcredentialv1.CredentialsRequest, error) {
+	stack := client.ObjectKey{Name: opts.BuildOpts.LokiStackName, Namespace: opts.BuildOpts.LokiStackNamespace}
+
+	providerSpec, secretName, err := encodeProviderSpec(opts.BuildOpts.LokiStackName, opts.ManagedAuthEnv)
+	if err != nil {
+		return nil, kverrors.Wrap(err, "failed encoding credentialsrequest provider spec")
+	}
+
+	return &cloudcredentialv1.CredentialsRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-%s", stack.Namespace, secretName),
+			Namespace: ccoNamespace,
+			Annotations: map[string]string{
+				AnnotationCredentialsRequestOwner: stack.String(),
```

Comment:
Question why do we use label and not something like the Owner reference https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/controller/controllerutil/controllerutil.go?

Reply:
As explained elsewhere we cannot use Kuberentes OwnerReferences for resources across namespaces or across scope (cluster-scope vs. namespaced).